subscribers.js

```diff
@@ -1,7 +1,90 @@
 "use strict";
 
-// Set of all registered emails.
-// TODO: Implement a real persistent storage solution.
-const gEmails = new Set();
+const pg = require("pg");
 
-module.exports = gEmails;
+const dbConfig = {
+  max: 10,
+};
+
+const dbPool = new pg.Pool(dbConfig);
+
+function dbq(aQuery) {
+  return dbPool.query(aQuery);
+}
+
+function _ret(aSuccess, aError, aOther) {
+  return Object.assign({
+    success: aSuccess,
+    error: aError || null,
+  }, aOther);
+}
+
+const Subscribers = {
+  async addUser(aEmail) {
+    // Error code for attempting to add a duplicate entry on a UNIQUE key.
+    const DUPLICATE_ERROR = 23505;
+    try {
+      await dbq({
+        text: "INSERT INTO users ( email ) VALUES ( $1 );",
+        values: [ aEmail ],
```

Comment:
Actually, looking at the db creation code, it feels like we may want to limit the email length here possibly (although I'm not 100% certain what happens when you try and write 4000 characters into a 320 character field in Postgres):

```js
await client.query("CREATE TABLE users ( email varchar(320) UNIQUE );");
```

... Or if we need to take unicode character length into consideration if people have email addresses w/ emojis or other valid glyphs.